bootstrap/v2/pkg/kfapp/kustomize/kustomize.go

```diff
@@ -577,44 +572,126 @@ func (kustomize *kustomize) writeConfigFile() error {
-	base := path.Join(kustomize.Spec.ManifestsRepo, compPath, "base")
-	baseKustomizationFile := filepath.Join(base, kftypes.KustomizationFile)
-	platformOverlay := path.Join(kustomize.Spec.ManifestsRepo, compPath, "overlays", kustomize.Spec.Platform)
-	platformKustomizationFile := filepath.Join(platformOverlay, kftypes.KustomizationFile)
-	if _, err := os.Stat(platformKustomizationFile); err == nil {
-		bases = append(bases, extractSuffix(kustomize.Spec.ManifestsRepo, platformOverlay))
-	} else if _, err := os.Stat(baseKustomizationFile); err == nil {
-		bases = append(bases, extractSuffix(kustomize.Spec.ManifestsRepo, base))
-	} else if _, err := os.Stat(compKustomizationFile); err == nil {
-		bases = append(bases, extractSuffix(kustomize.Spec.ManifestsRepo, comp))
+func (kustomize *kustomize) overlayResources(compName string, compPath string) map[string]*types.Kustomization {
+	params := kustomize.Spec.ComponentParams[compName]
+	overlays := []string{}
+	if params != nil {
+		for _, nv := range params {
+			name := nv.Name
+			if name == "overlay" {
+				overlays = append(overlays, nv.Value)
+			}
+		}
 	}
+	resources := func(compDir string, overlayPath string) *types.Kustomization {
+		kustomizationFile := filepath.Join(overlayPath, kftypes.KustomizationFile)
+		data, err := ioutil.ReadFile(kustomizationFile)
+		if err != nil {
+			return nil
+		}
+		kustomization := &types.Kustomization{
+			TypeMeta: types.TypeMeta{
+				APIVersion: types.KustomizationVersion,
+				Kind:       types.KustomizationKind,
+			},
+			Namespace: kustomize.Namespace,
+			Bases: []string{},
+			PatchesStrategicMerge: []patch.StrategicMerge{},
+			PatchesJson6902: []patch.Json6902{},
+		}
+		overlayKustomization := &types.Kustomization{}
+		if err = yaml.Unmarshal(data, overlayKustomization); err != nil {
+			return nil
+		}
+		if overlayKustomization.Bases != nil {
+			for _, value := range overlayKustomization.Bases {
+				baseAbsolutePath := path.Join(overlayPath, value)
+				basePath := extractSuffix(compDir, baseAbsolutePath)
+				kustomization.Bases = append(kustomization.Bases, basePath)
+			}
+		}
+		if overlayKustomization.PatchesStrategicMerge != nil {
+			for _, value := range overlayKustomization.PatchesStrategicMerge {
+				patchAbsoluteFile := filepath.Join(overlayPath, string(value))
+				patchFile := extractSuffix(compDir, patchAbsoluteFile)
+				patchFileCasted := patch.StrategicMerge(patchFile)
+				kustomization.PatchesStrategicMerge = append(kustomization.PatchesStrategicMerge, patchFileCasted)
+			}
+		}
+		if overlayKustomization.PatchesJson6902 != nil {
+			for _, value := range overlayKustomization.PatchesJson6902 {
+				patchJson := new(patch.Json6902)
+				patchJson.Target = value.Target
+				patchAbsolutePath := filepath.Join(overlayPath, value.Path)
+				patchJson.Path = extractSuffix(compDir, patchAbsolutePath)
+				kustomization.PatchesJson6902 = append(kustomization.PatchesJson6902, *patchJson)
+			}
+		}
```

Comment:
`configMapGenerator`, `secretGenerator`,`crds` in overlay also creates k8s resources.
In addition, overlay resources may refer `vars`.

Reply:
Updated